### PR TITLE
Tickets/sitcom 946

### PR DESCRIPTION
**Pull request by ykangastro**
This is a PR for technoten-082. I wanted to add Petr as a reviewer, but I couldn't find his name on the list of Reviewers for lsst-sticom. I might update or change minor things more tonight if necessary.

### CODE REVIEW

File: index.rst [hunk 1]
```diff
@@ -6,24 +6,219 @@
...
 
-.. note::
-
-   **This technote is a work-in-progress.**
-
 Abstract
 ========
 
-This is a technote for the description of results from Individual Hardpoint Breakaway Test. 
+This technote describes the analysis and results derived from individual hard point breakaway tests conducted with the M1M3 surrogate.
```
Comment (b1quint) [inline index.rst:12]: I would convert this into a link to the next session.

File: index.rst [hunk 2]
```diff
@@ -6,24 +6,219 @@
...
+The notebook is located within `notebooks_vandv <https://github.com/lsst-sitcom/notebooks_vandv/blob/tickets/SITCOM-838/notebooks/tel_and_site/subsys_req_ver/m1m3/SITCOM-838_Anaysis.ipynb>`_ GitHub repository.
+With all the steps guarded with reasonable timeouts, so problems are detected if hard point cannot travel to reach low or high limit switches, etc.
+**If this test shows that the hard points do not work properly at the limits, this could be one of the blockers for the installation of M1M3 until it is solved.** 
+
+
+Hardpoint Breakaway Test
+========================
+
+The active support system of the M1M3 includes six axial hard point actutators in the hexapod configuration. :cite:`2018SPIE10700E..3GD`
```
Comment (b1quint) [inline index.rst]: in the hexapod -> in a hexapod ?

File: index.rst [hunk 3]
```diff
@@ -6,24 +6,219 @@
...
+These hard point actuators should minimize forces during slews at any TMA position and be kept under the breakaway limit.
```
Comment (b1quint) [inline index.rst:23]: I would confirm this sentence with Petr. From what I understand, the 156 Force Actuators are responsible for keeping the forces in the Hard Points small. The HPs are only responsible for guiding the rigid body motion of the mirror, and the forces applied to them should be within the range you describe later. It might be my misunderstanding, but this is a good opportunity to review and clarify it.

File: index.rst [hunk 4]
```diff
@@ -6,24 +6,219 @@
...
+
+Hardpoint Breakaway Test
+========================
+
+The active support system of the M1M3 includes six axial hard point actutators in the hexapod configuration. :cite:`2018SPIE10700E..3GD`
+These hard point actuators should minimize forces during slews at any TMA position and be kept under the breakaway limit. 
+The breakaway limits for each hard points should happen in the range of -4420 to -3456 N for retraction and 2981 to 3959 N for extension.
+The followings steps are performed during the individual hard point breakaway test.
+
+1. Move hard point in negative (increasing tension) direction until either a low limit switch is actuated
```
Comment (b1quint) [inline index.rst]: I don't think that we have a "limit switch." We might want to clarify a bit more what is a "break away" in this context. I would review the "breakaway" definition with Doug, but I believe it is when the Hard Point Break Away system receives a force so high that it "gives up."

Comment (b1quint) [inline index.rst]: remove "either"

File: index.rst [hunk 5]
```diff
@@ -6,24 +6,219 @@
...
+
+2. Move hard point in positive (increasing compression) direction till the high limit switch is triggered
```
Comment (b1quint) [inline index.rst:29]: Same as above

File: index.rst [hunk 6]
```diff
@@ -6,24 +6,219 @@
...
+
+3. Move hard point downwards (increasing tension) until the low limit switch is hit
+
+4. Move hard point back to reference position
+
+5. Wait for the hard point to reach the reference position
+
+
+Note that for tests only full travel, from low limit to high limit and back, performed in steps 2 and 3 shall be considered. Other hard point movements do not guarantee a full hard point motion range will be examined.
```
Comment (b1quint) [inline index.rst]: I understand what you are saying, but I would reword the first sentence in this paragraph for a clearer message.

File: index.rst [hunk 7]
```diff
@@ -6,24 +6,219 @@
...
+    | 40       | 153    | 2023-05-26T02:23:28  | 100034   |
+    +----------+--------+----------------------+----------+ 
+    | 89.95    | 153    | 2023-06-20T03:11:00  | 100038   |
+    +----------+--------+----------------------+----------+ 
+    | 90       | -29.69 | 2023-05-31T05:44:14  | 100061   |  
+    +----------+--------+----------------------+----------+ 
+
+
+
+Hard Point Breakaway Test at el=90
```
Comment (b1quint) [inline index.rst]: Consider having a section called "Hard Point Breakaway Tests Results" or simply "Tests Results." This session should contain the results for each of the test runs you present. Something like

- Test Results
    - HP Test at el 90 deg
    - HP Test at el 0 deg
    - ...

File: index.rst [hunk 8]
```diff
@@ -6,24 +6,219 @@
...
 ========
 
-This is a technote for the description of results from Individual Hardpoint Breakaway Test. 
+This technote describes the analysis and results derived from individual hard point breakaway tests conducted with the M1M3 surrogate.
+The notebook is located within `notebooks_vandv <https://github.com/lsst-sitcom/notebooks_vandv/blob/tickets/SITCOM-838/notebooks/tel_and_site/subsys_req_ver/m1m3/SITCOM-838_Anaysis.ipynb>`_ GitHub repository.
+With all the steps guarded with reasonable timeouts, so problems are detected if hard point cannot travel to reach low or high limit switches, etc.
+**If this test shows that the hard points do not work properly at the limits, this could be one of the blockers for the installation of M1M3 until it is solved.** 
+
+
+Hardpoint Breakaway Test
```
Comment (b1quint) [inline index.rst:19]: I missed here a discussion associated with the stiffness. I know it is a bit obscure and I have been asking Doug to clarify. Here is a message that I got from him in an e-mail:

> When we do a breakaway test the actual length of the hardpoint does not change. However, the displacement mechanisms is operated to apply the tension/compression on the breakaway mechanism. The displacement of the mechanism has to match the inverse of the displacement of the breakaway mechanism. They have to be equal and opposite to retain the same overall length.
>  
> When the breakaway is locked and the load is changed, up there is actually as small change in the overall length due to elastic deformation of the overall system. This provides the slant in the force displacement curve.
>  
> Although the actuator does not actually change length, the actuator thinks it is changing length due to the operation of its displacement mechanism. It reports that it is changing length. A separate LVDT measures the length change of the breakaway mechanism. This LVDT is not presently working correctly. It is reporting both the breakaway LVDT and the displacement actuator LVDT. When Both LFDT are both working correctly, they should be equal and opposite, except over the elastic deformation range.

Ideally, I would ask to have a reference to a document that explains this with details (ask Doug?) and have one paragraph trying to explain it here. I know it is not super easy, and many ppl have difficulties understanding it.

File: index.rst [hunk 9]
```diff
@@ -6,24 +6,219 @@
...
+    +----------+--------+----------------------+----------+ 
+    | 90       | -29.69 | 2023-05-31T05:44:14  | 100061   |  
+    +----------+--------+----------------------+----------+ 
+
+
+
+Hard Point Breakaway Test at el=90
+==================================
+
+These are results from hard point breakaway tests when the TMA is positioned at el=90 deg, az=-29.69 deg.
```
Comment (b1quint) [inline index.rst:103]: Could you add a summary or interpretation of the test results here? Something like: they all look good at 90º because of X, Y, and Z. Similarly, if they fail, explain quickly why.

File: index.rst [hunk 10]
```diff
@@ -6,24 +6,219 @@
...
+3. Move hard point downwards (increasing tension) until the low limit switch is hit
+
+4. Move hard point back to reference position
+
+5. Wait for the hard point to reach the reference position
+
+
+Note that for tests only full travel, from low limit to high limit and back, performed in steps 2 and 3 shall be considered. Other hard point movements do not guarantee a full hard point motion range will be examined.
+
+
```
Comment (b1quint) [inline index.rst:54]: Can you add a new section here with a list of requirements that we are trying to verify here? If you don't have the list right away, you can create a session and create a todo box as a reminder to do this in a new PR. Something like

```
Requirements and Tickets
===================

Find here the JIRA tickets and requirements associated with this test. 

.. todo::

    Add associated requirements to this technote.
```

Comment (b1quint) [reply index.rst:54]: If you get the requirements quickly enough, simply add them. But consider adding them as part of a new ticket/PR.

File: index.rst [hunk 11]
```diff
@@ -6,24 +6,219 @@
...
+
+
+
+Hard Point Breakaway Test at El=0
+===================================
+
+These are results from hard point breakaway test when the TMA was positioned at el=0 deg, az=-29.69 deg. 
+In Figure 4, hard point 2 and hard point 5 were not moving to the positive direction.
```
Comment (b1quint) [inline index.rst:147]: This is a good example of a test result summary. 👍🏽

File: index.rst [hunk 12]
```diff
@@ -6,24 +6,219 @@
...
+
+
+
+Hard Point Breakaway Test at El=0
+===================================
+
+These are results from hard point breakaway test when the TMA was positioned at el=0 deg, az=-29.69 deg. 
+In Figure 4, hard point 2 and hard point 5 were not moving to the positive direction. 
+Hardpoint 1 and hard point 6 were both staying on the position for testing positive for a shorter period of time whereas hard point 3 and hard point 4 were staying on testing negative position for a shorter period time.  
+This is because depending on the position of each hard point.
```
Comment (b1quint) [inline index.rst:149]: We need a reference that shows the position of the hardpoints in the introduction and that can be referenced here as well. Again, if you find it, add it here. If it will take too long to find the info, add another `.. todo` here and in the intro.

File: index.rst [hunk 13]
```diff
@@ -6,24 +6,219 @@
...
+   
+   Figure 6. Same as Figure 3 but the TMA at el=0 deg.   
+
+
+Hard Point Breakaway Test at el=40
+===================================
+
+.. figure:: /_static/m1m3004_hp_timeline_El_40.png
+   
+   Figure 7. Same as Figure 1 but the TMA at el=40 deg.
```
Comment (b1quint) [inline index.rst]: I know it can be a bit repetitive, but imagine someone reading this document and having to scroll up and down every time to remember what was in the description of the plot. I suggest you repeat the explanation on each of them.

File: index.rst [hunk 14]
```diff
@@ -6,24 +6,219 @@
...
+===================================
+
+These are results from hard point breakaway test when the TMA was positioned at el=0 deg, az=-29.69 deg. 
+In Figure 4, hard point 2 and hard point 5 were not moving to the positive direction. 
+Hardpoint 1 and hard point 6 were both staying on the position for testing positive for a shorter period of time whereas hard point 3 and hard point 4 were staying on testing negative position for a shorter period time.  
+This is because depending on the position of each hard point.  
+
+.. figure:: /_static/m1m3004_hp_timeline_El_0.png
+   
+   Figure 4. Same as Figure 1 but the TMA at el=0 deg.
```
Comment (b1quint) [inline index.rst]: "Figure 4" appears duplicated in the HTML version.

File: index.rst [hunk 15]
```diff
@@ -6,24 +6,219 @@
...
+These are results from hard point breakaway test when the TMA was positioned at el=0 deg, az=-29.69 deg. 
+In Figure 4, hard point 2 and hard point 5 were not moving to the positive direction. 
+Hardpoint 1 and hard point 6 were both staying on the position for testing positive for a shorter period of time whereas hard point 3 and hard point 4 were staying on testing negative position for a shorter period time.  
+This is because depending on the position of each hard point.  
+
+.. figure:: /_static/m1m3004_hp_timeline_El_0.png
+   
+   Figure 4. Same as Figure 1 but the TMA at el=0 deg. 
+
+The stiffness of each curves are fitted from :math:`\Delta`\displacement = 0 :math:`{\mu}m` (Figure 5).
```
Comment (b1quint) [inline index.rst:160]: Same for Figure 5. I would review other figures.

File: index.rst [hunk 16]
```diff
@@ -6,24 +6,219 @@
...
+    +----------+--------+----------------------+----------+ 
+    | 40       | 153    | 2023-05-26T02:23:28  | 100034   |
+    +----------+--------+----------------------+----------+ 
+    | 89.95    | 153    | 2023-06-20T03:11:00  | 100038   |
+    +----------+--------+----------------------+----------+ 
+    | 90       | -29.69 | 2023-05-31T05:44:14  | 100061   |  
+    +----------+--------+----------------------+----------+ 
+
+
+
```
Comment (b1quint) [inline index.rst:88]: Considering that we have lots of plots and results below, I would consider adding a new section with a general analysis of the behavior of the HPs during the test. Again, I would create the session and add a `.. todo` here so you can do this in a separate PR. Of course, you and do it here if you think it is faster.

File: index.rst (hunk 4 above)
Comment (ykangastro) [reply index.rst]: I already forwarded the response from Petr to you but those steps were from Petr. But let's put it on the To-do list to check with others. 

> 1. Move hardpoint in negative (increasing tension) direction until either a low limit switch is actuated
> 2. Move hardpoint in positive (increasing compression) direction till the high limit switch is triggered
> 3. Move hardpoint downwards (increasing tension) until the low limit switch is hit
> 4. Move hardpoint back to reference position
> 5. Wait for the hardpoint to reach the reference position

File: index.rst (hunk 8 above)
Comment (ykangastro) [reply index.rst:19]: I guess that we can check it from Felipe's SPIE paper [ads](https://ui.adsabs.harvard.edu/abs/2018SPIE10700E..3GD/abstract) (sorry I couldn't read it thoroughly). If you need the paper in MS Word file format, let me know. I can share with you. I will put this on the "To-do list"

File: index.rst (hunk 10 above)
Comment (ykangastro) [reply index.rst:54]: >  LTS-88    LTS-88-REQ-0018-V-01: 3.7.5.1 Load Limiting Axial Breakaway Mechanism Displacement
> LVV-11200 LTS-88-REQ-0015-V-01: 3.7.1.3 Hardpoint Displacement Repeatability and Resolution_1
> LVV-11184 LTS-88-REQ-0024-V-01: 3.7.1.7 Hardpoint Axial Breakaway Repeatability_1
> LVV-11208 LTS-88-REQ-0025-V-01: 3.7.1.8 Hardpoint Stiffness Limits_1

File: index.rst (hunk 16 above)
Comment (ykangastro) [reply index.rst:88]: A quick question - What should "a general analysis" of the behavior of the HPs during the test include? Maybe I can put and compare the results from el=90deg and el=0deg for this section (or any ideas?)

Let me put this on the todo

File: index.rst (hunk 15 above)
Comment (ykangastro) [reply index.rst:160]: This would be a kind of technical issue with build HTML (it was fine with local index.html). Let me check and try to fix it.

File: index.rst (hunk 9 above)
Comment (ykangastro) [reply index.rst:103]: I put "Measured forces on all of hard points look working properly because breakaway happened in the range of the requirement (Tension: -4420 - -3456N, Compression: 2981 - 3959N)." at the end of this sentence. But I think we'd better put more detailed explanations or descriptions in the section for the "A general analysis" later.

File: index.rst (hunk 6 above)
Comment (ykangastro) [reply index.rst]: I think "Note that" parts can be removed for now. I guess those sentences are written for my reference by Petr.

**Review by b1quint — APPROVED** [review]
We talked and agreed to complete the ToDo tasks in separate PRs.
